pkg/controllerutils/miscellaneous.go

```diff
@@ -124,3 +128,25 @@ func ReconcileOncePer24hDuration(clock clock.Clock, objectMeta metav1.ObjectMeta
 
 	return 0
 }
+
+// GetMainReconciliationContext returns a context with timeout for the controller's main client. The resulting context has a timeout equal to the timeout passed in the argument but
+// not more than DefaultReconciliationTimeout.
+func GetMainReconciliationContext(ctx context.Context, timeout time.Duration) (context.Context, context.CancelFunc) {
```

Comment:
Can you add unit tests for the new utility functions?

Reply:
This is such a simple function, why and how would you test it? Where's the benefit in the hassle?

Reply:
The unit test would have been to make sure that the returned context is properly limited to the `DefaultReconciliationTimeout` and for the second function - that the returned context's timeout is also halved. However, I guess you are right that trying to compare the deadlines can be a hassle (and maybe somewhat flaky). So you can disregard this review comment.